src/Resolvers/DataValidationRulesResolver.php

```diff
@@ -278,4 +289,9 @@ protected function inferRulesForDataProperty(
             $path
         );
     }
+
+    protected function shouldMergeRules(DataClass $class): bool
+    {
+        return $class->attributes->contains(fn (object $attribute) => $attribute::class === MergeRules::class);
```

Comment:
No need for an extra function here, just inline the variable

Reply:
Have removed this